design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
+- This opens a dialog from which the user can set all supported constraint types.
+- The user sees a layout with two panels. On one side, all the existing constraints are listed. On the other side, a form with the constraints configuration is displayed. The form corresponds to the currently active list item.
+- The list panel contains actions at the bottom of the panel. From there, the user can choose to add or delete an item from the list. The user clicks on 'Add' to create a new constraint.
+- The user is presented with a form. From there, they select the type of constraint to be added. The user selects the 'Foreign Key' constraint option from the list.
+- The user selects single or multiple columns they wish to apply the constraint to.
+- The user selects a table to be referenced by these columns.
+- The user selects a column in the reference table from which the column will match values. By default, this is set to the 'Primary Key' column in the referenced table. When changing this field, the user can select only columns with a UNIQUE or PRIMARY KEY constraint.
 
----
+## Scenario 2: User Edits an Existing Foreign Key Constraint
+
+### Scenario 2a: From the column header menu
+
+#### Steps for 2a
+
+- The user hovers over the foreign key reference link below the column name.
+- Information about the constraints applied to the column is displayed in a tooltip.
+- The user clicks on the foreign key reference link below the column name.
+- Constraint settings open for the column.
+
+### Scenario 2b: From table constraint settings
+
+#### Steps for 2b
+
+- A user wants to edit an existing foreign key constraint to change the selected columns or select another referenced table.
+- The user can open a specific foreign key constraint form by clicking on the foreign key link in the column header.
+- If the modified FK constraint causes existing column values to become invalid, the user will receive a warning and be aware of the potential implications of this change, like loss of data or broken views.
+
+## Scenario 3: User Identifies a Column With a Foreign Key Constraint Applied
+
+### Scenario 3a: The foreign key constraint is set to a single column
+
+#### Steps for 3a
+
+- The user opens a table that contains columns with a foreign key constraint applied.
+- The user identifies the columns in two ways:
+  - The user looks at the column header and sees the foreign key indicator. This indicator shows a key icon and the referenced table and column next to it.
+    - The user can click the indicator to display the foreign key settings for the specific column.
+    - The indicator can be hovered over to display a tooltip with information about the foreign key constraints.
+  - The user looks at the column fields and sees that values are added with different styling, a tag-like element with a colored background.
```

Comment:
In the loom you mentioned the idea of using different colors for different tables. That's a neat idea, but I'd prefer we skip that at first for the sake of simplicity.

Using dynamic colors introduces some additional UX complexities such as

- How do we theme those colors for light-mode/dark-mode?
- How do we ensure the colors are sufficiently distinct?
- If tables A and B both have FKs to table C, then do they they both use the same color for table C?
- Can the user configure the colors?

All of these questions are solvable of course, but it's low-priority compared to the other tasks we have, so I say let's do it later.

Reply:
I think these can be on the spec, but we can discuss and de-prioritize them when working on the issues. We could probably link this comment to the issue, it's a conversation better placed there.

Reply:
I agree that we can keep them in the spec. I'll make sure to create a separate issue for implementing the colors that we can defer if it makes sense.

Reply:
Agree, I'll keep this in the spec. We can define how we want to apply color once design for Views is complete.